keda/templates/manager/minimal-rbac.yaml

```diff
@@ -0,0 +1,120 @@
+{{- if .Values.rbac.create }}
+apiVersion: rbac.authorization.k8s.io/v1
+kind: Role
+metadata:
+  {{- with .Values.additionalAnnotations }}
+  annotations:
+    {{- toYaml . | nindent 4 }}
+  {{- end }}
+  labels:
+    app.kubernetes.io/name: {{ .Values.operator.name }}-certs
+    {{- include "keda.labels" . | indent 4 }}
+  name: {{ .Values.operator.name }}-certs
+  namespace: {{ .Release.Namespace }}
+rules:
+- apiGroups:
+  - coordination.k8s.io
+  resources:
+  - leases
+  verbs:
+  - '*'
+{{- if and .Values.certificates.autoGenerated (not .Values.certificates.certManager.enabled) }}
+- apiGroups:
+  - ""
+  resources:
+  - secrets
+  verbs:
+  - get
+  resourceNames:
+  - {{ .Values.certificates.secretName | quote }}
+- apiGroups:
+  - ""
+  resources:
+  - secrets
+  verbs:
+  - create
+  - update
+{{- end }}
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: RoleBinding
+metadata:
+  {{- with .Values.additionalAnnotations }}
+  annotations:
+    {{- toYaml . | nindent 4 }}
+  {{- end }}
+  labels:
+    app.kubernetes.io/name: {{ .Values.operator.name }}-certs
+    {{- include "keda.labels" . | indent 4 }}
+  name: {{ .Values.operator.name }}-certs
+  namespace: {{ .Release.Namespace }}
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: Role
+  name: {{ .Values.operator.name }}-certs
+subjects:
+- kind: ServiceAccount
+  name: {{ (.Values.serviceAccount.operator).name | default .Values.serviceAccount.name }}
+  namespace: {{ .Release.Namespace }}
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  {{- with .Values.additionalAnnotations }}
+  annotations:
+    {{- toYaml . | nindent 4 }}
+  {{- end }}
+  labels:
+    app.kubernetes.io/name: {{ .Values.operator.name }}-minimal-cluster-role
+    {{- include "keda.labels" . | indent 4 }}
+  name: {{ .Values.operator.name }}-minimal-cluster-role
+rules:
+- apiGroups:
+  - keda.sh
+  resources:
+  - clustertriggerauthentications
+  verbs:
+  - '*'
```

Comment:
I think that this permissions is already granted in the operator rbac

Reply:
you are right, but I'd leave it here and remove it from the other place. Because these 2 `(Cluster)Role+(Cluster)RoleBinding` are the bare minimum that is needed for operator to even start. The `clustertriggerauthentications` is a cluster-wide CRD so it doesn't make sense to include it in the "`watch_namespace` logic".